src/snowflake/connector/backoff_policies.py

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import random
+from typing import Callable, Iterator
+
+"""This module provides common implementations of backoff policies
+
+All backoff policies must be implemented as generator functions with the behaviour specified below. These generator
+functions will be called to create iterators yielding backoff durations.
+
+Args:
+    None
+
+Yields:
+    int: Next backoff duration in seconds
+
+Example:
+    This is an example of a valid backoff policy that always yields a backoff duration of 42 seconds.
+
+    def constant_backoff() -> int:
+        while True
+            yield 42
+
+
+Note:
+    The functions provided in this module are not backoff policies. They are functions returning backoff policies.
+    This is to enable customization of the constants used in backoff computations.
+"""
+
+DEFAULT_BACKOFF_FACTOR = 2
+DEFAULT_BACKOFF_BASE = 1
+DEFAULT_BACKOFF_CAP = 16
+DEFAULT_ENABLE_JITTER = True
+
+
+def mixed_backoff(
+    factor: int = DEFAULT_BACKOFF_FACTOR,
+    base: int = DEFAULT_BACKOFF_BASE,
+    cap: int = DEFAULT_BACKOFF_CAP,
+    enable_jitter: bool = DEFAULT_ENABLE_JITTER,
+) -> Callable[..., Iterator[int]]:
```

Comment:
`time.sleep` takes a [float](https://github.com/python/typeshed/blob/ba3f23b2040927cb278dcb4d5ea5f8c43eb5c8b3/stdlib/time.pyi#L75). Maybe these should all return floats